serial_monitor_to_file.py

```diff
@@ -0,0 +1,46 @@
+#imports
+import serial.tools.list_ports
+from datetime import datetime
+
+#list out all ports in use
+ports = serial.tools.list_ports.comports()
+serial_instance = serial.Serial()
+port_list = []
+port_variable = ""
+
+for port in ports:
+    #append each port to the list of ports
+    port_list.append(port)
+    #print out all ports
+    print(port)
+
+
+value = input("select Port: COM")
+#searches the list of ports matching the one picked
+for i in range(0, len(port_list)):
+    if str(port_list[i]).startswith("COM" +value):
+        port_variable = "COM" + value
+        print(port_list[i])
+
+#must be the same baudrate as arduino
+serial_instance.baudrate = 9600
+serial_instance.port = port_variable
+serial_instance.open()
+
+#Creat a new file to steam serial data to.
+file = open(datetime.now()+".txt", "w")
+
+while 1:
+    arduino_data = serial_instance.readline()
+    #if we read STOP we stop streaming data to file
+    if 'STOP' in str(arduino_data.decode()):
+        print("A stop have been read: The program will now close and save the file under name: ")
+        k = input("press enter to close")
+        break
+    
+    #save line to file, with decoded from byte to string.
+    print(arduino_data.decode().rstrip())
+    file.write(arduino_data.decode().rstrip())
```

Comment:
Jeg ville nok foreslå at skrive det til filen mere sjældent end hver eneste gang den receiver noget.
Der er to muligheder. Enten skriver vi til en buffer, som bliver gemt til filen efter en bestemt længde.
Eller også så gemmer vi først filen, når vi modtager et STOP.

Hvis ikke vi gør sådan, så skriver vi alt for ofte til disken.

Reply:
hmm ima cook something up.